crypto/rsa/rsa_oaep.c

```diff
@@ -73,9 +73,23 @@ int ossl_rsa_padding_add_PKCS1_OAEP_mgf1_ex(OSSL_LIB_CTX *libctx,
         return 0;
 #endif
     }
+
+#ifdef FIPS_MODULE
+    if (EVP_MD_is_a(md, "SHAKE-128") || EVP_MD_is_a(md, "SHAKE-256")) {
+        ERR_raise(ERR_LIB_RSA, RSA_R_DIGEST_NOT_ALLOWED);
+        return 0;
+    }
+#endif
```

Comment:
I'd suggest wrapping this into a utility function to save the repetition.

Reply:
I agree. Do I correctly understand that this change is acceptable after refactoring?

Reply:
Best wait until after the OTC call.

Reply:
Sure, now I'm trying to summarize what is **not** acceptable